File: backend/internal/models/expense.go
package models

import (
	"time"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

// ExpenseCategory represents a category for classifying expenses
type ExpenseCategory struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description *string   `json:"description,omitempty"`
	IsActive    bool      `json:"is_active"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// Expense represents an operating expense
type Expense struct {
	ID              uuid.UUID        `json:"id"`
	CategoryID      *uuid.UUID       `json:"category_id,omitempty"`
	CategoryName    *string          `json:"category_name,omitempty"`
	ProductID       *uuid.UUID       `json:"product_id,omitempty"`
	ProductName     *string          `json:"product_name,omitempty"`
	Quantity        *decimal.Decimal `json:"quantity,omitempty"`
	Amount          decimal.Decimal  `json:"amount"`
	Description     string           `json:"description"`
	ExpenseDate     time.Time        `json:"expense_date"`
	Vendor          *string          `json:"vendor,omitempty"`
	ReferenceNumber *string          `json:"reference_number,omitempty"`
	Notes           *string          `json:"notes,omitempty"`
	CreatedBy       uuid.UUID        `json:"created_by"`
	CreatedByName   *string          `json:"created_by_name,omitempty"`
	CreatedAt       time.Time        `json:"created_at"`
	UpdatedAt       time.Time        `json:"updated_at"`
}

// ExpenseSummary represents aggregated expense data
type ExpenseSummary struct {
	TotalAmount  decimal.Decimal          `json:"total_amount"`
	ExpenseCount int                      `json:"expense_count"`
	ByCategory   []CategoryExpenseSummary `json:"by_category"`
}

// CategoryExpenseSummary represents expense totals by category
type CategoryExpenseSummary struct {
	CategoryID   *uuid.UUID      `json:"category_id,omitempty"`
	CategoryName string          `json:"category_name"`
	TotalAmount  decimal.Decimal `json:"total_amount"`
	Count        int             `json:"count"`
}

File: frontend/src/types/expense.ts
export interface ExpenseCategory {
  id: string;
  name: string;
  system_key?: string;
  description?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Expense {
  id: string;
  category_id?: string;
  category_name?: string;
  product_id?: string;
  product_name?: string;
  quantity?: string;
  applies_inventory: boolean;
  amount: string;
  description: string;
  expense_date: string;
  vendor?: string;
  reference_number?: string;
  notes?: string;
  created_by: string;
  created_by_name?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateExpenseRequest {
  category_id?: string;
  product_id?: string;
  quantity?: string;
  applies_inventory?: boolean;
  amount: string;
  description: string;
  expense_date: string;
  vendor?: string;
  reference_number?: string;
  notes?: string;
}

export interface UpdateExpenseRequest {
  category_id?: string;
  product_id?: string;
  quantity?: string;
  applies_inventory?: boolean;
  amount?: string;
  description?: string;
  expense_date?: string;
  vendor?: string;
  reference_number?: string;
  notes?: string;
  expected_updated_at?: string;
}

export interface ExpenseSummary {
  total_amount: string;
  expense_count: number;
  by_category: CategoryExpenseSummary[];
  start_date: string;
  end_date: string;
}

export interface CategoryExpenseSummary {
  category_id?: string;
  category_name: string;
  total_amount: string;
  count: number;
}
